pkg/tcpip/stack/ndp.go

```diff
@@ -720,6 +743,83 @@ func (ndp *ndpState) handleRA(ip tcpip.Address, ra header.NDPRouterAdvert) {
 	}
 }
 
+func (ndp *ndpState) sendRouterAdvert() {
+	ref := ndp.nic.getRefOrCreateTemp(header.IPv6ProtocolNumber, header.IPv6AllRoutersMulticastAddress, NeverPrimaryEndpoint, forceSpoofing)
+	la := tcpip.Address(net.ParseIP("fe80::2")) //
+	r := makeRoute(header.IPv6ProtocolNumber, la, header.IPv6AllNodesMulticastAddress, ndp.nic.linkEP.LinkAddress(), ref, false, false)
+	defer r.Release()
+
+	// Route should resolve immediately since
+	// header.IPv6AllRoutersMulticastAddress is a multicast address so a
+	// remote link address can be calculated without a resolution process.
+	if c, err := r.Resolve(nil); err != nil {
+		log.Fatalf("ndp: error when resolving route to send NDP RA (%s -> %s on NIC(%d)): %s", header.IPv6Any, header.IPv6AllRoutersMulticastAddress, ndp.nic.ID(), err)
+	} else if c != nil {
+		log.Fatalf("ndp: route resolution not immediate for route to send NDP RA (%s -> %s on NIC(%d))", header.IPv6Any, header.IPv6AllRoutersMulticastAddress, ndp.nic.ID())
+	}
+
+	pi := header.NewNDPPrefixInformation()
+	pi.SetPrefix(ndp.configs.routerConfig.Prefix)
+	pi.SetPrefixLength(ndp.configs.routerConfig.PrefixLength)
+	pi.SetOnLinkFlag(ndp.configs.routerConfig.OnLinkFlag)
+	pi.SetAutonomousAddressConfigurationFlag(ndp.configs.routerConfig.AutonomousAddressConfigurationFlag)
+	pi.SetPreferredLifetime(ndp.configs.routerConfig.PreferredLifetime)
+	pi.SetValidLifetime(ndp.configs.routerConfig.ValidLifetime)
+
+	optsSerializer := header.NDPOptionsSerializer{
+		header.NDPSourceLinkLayerAddressOption(r.LocalLinkAddress[:]),
+		pi,
+	}
+	hdr := buffer.NewPrependable(int(r.MaxHeaderLength()) + header.ICMPv6RouterAdvertMinimumSize + int(optsSerializer.Length()))
```

Comment:
```
```suggestion
	icmpv6RASize := header.ICMPv6RouterAdvertMinimumSize + int(optsSerializer.Length())
	hdr := buffer.NewPrependable(int(r.MaxHeaderLength()) + header.ICMPv6RouterAdvertMinimumSize + icmpv6RASize)
	packet := header.ICMPv6(hdr.Prepend(icmpv6RASize))
```